crates/query-engine/translation/src/translation/query/values.rs

```diff
@@ -84,48 +84,88 @@ fn type_to_ast_scalar_type(typ: &database::Type) -> sql::ast::ScalarType {
 
 /// Convert a variable into a SQL value.
 pub fn translate_variable(
+    state: &mut State,
     variables_table: sql::ast::TableReference,
     variable: String,
     r#type: &database::Type,
 ) -> sql::ast::Expression {
-    let variables_reference =
-        Expression::ColumnReference(sql::ast::ColumnReference::AliasedColumn {
-            table: variables_table,
-            column: sql::helpers::make_column_alias(sql::helpers::VARIABLES_FIELD.to_string()),
-        });
+    let variables_reference = Expression::ColumnReference(ColumnReference::AliasedColumn {
+        table: variables_table,
+        column: sql::helpers::make_column_alias(sql::helpers::VARIABLES_FIELD.to_string()),
+    });
 
-    match r#type {
-        database::Type::CompositeType(_type_name) => {
-            let exp = sql::ast::Expression::BinaryOperation {
+    let projected_variable_exp = match r#type {
+        database::Type::CompositeType(_) | database::Type::ArrayType(_) => {
+            sql::ast::Expression::BinaryOperation {
                 left: Box::new(variables_reference),
                 operator: sql::ast::BinaryOperator("->".to_string()),
                 right: Box::new(sql::ast::Expression::Value(sql::ast::Value::String(
                     variable.clone(),
                 ))),
-            };
-            sql::ast::Expression::FunctionCall {
-                function: sql::ast::Function::JsonbPopulateRecord,
-                args: vec![
-                    sql::ast::Expression::Cast {
-                        expression: Box::new(sql::ast::Expression::Value(sql::ast::Value::Null)),
-                        r#type: type_to_ast_scalar_type(r#type),
-                    },
-                    exp,
-                ],
             }
         }
-        _ => {
-            let exp = sql::ast::Expression::BinaryOperation {
-                left: Box::new(variables_reference),
-                operator: sql::ast::BinaryOperator("->>".to_string()),
-                right: Box::new(sql::ast::Expression::Value(sql::ast::Value::String(
-                    variable.clone(),
-                ))),
+        database::Type::ScalarType(_) => sql::ast::Expression::BinaryOperation {
+            left: Box::new(variables_reference),
+            operator: sql::ast::BinaryOperator("->>".to_string()),
```

Comment:
```suggestion
            // For scalar types, we use `->>` to return the value as `text`
            // because ...
            operator: sql::ast::BinaryOperator("->>".to_string()),
```